### ISSUE TITLE
[Bug] Codegen: packed prefill >128 tokens emits NaN logits since 7542c36d (explicit-tmp tcvt/tmrgsort + PTOAS v0.60)

### ISSUE THREAD

**Issue by ndleslx** [hw-native-sys/pypto#2558]
## Background

In pypto-lib, `models/deepseek_v4_flash_mtp/prefill_fwd.py` (`l3_prefill_fwd`, the packed-prefill program DeepSeek-V4-Flash serving uses) on `a2a3` (8 devices) produces **NaN logits for any packed prefill extent above 128 tokens with real W8A8 weights** (extents 64/128 generate correctly; 129/160/256/302 fail deterministically). Full symptom details, extent table, and the follow-on AICore fault chain are in pypto-lib#1072 — that issue also shows pypto-lib's own standalone harness hides the NaNs (`[RUN] PASS (validation skipped)`).

A pypto-commit bisect of that failure (green `42881d7b` → red `57e5c315`) now localizes it:

| # | pypto | simpler | ptoas | Result |
|---|---|---|---|---|
| 1 | `42881d7b` | `799640e6` | 0.57 | ✅ correct greedy generation |
| 2 | `8cf3f058` (= 7542c36d's parent) | `799640e6` | 0.57 | ✅ correct greedy generation |
| 3 | `8cf3f058` | `0659e29b` | 0.57 | ✅ correct — simpler bump exonerated |
| 4 | `8cf3f058` | `799640e6` | 0.60 | ✗ assemble error: `'pto.tcvt' op requires explicit tmp ...` |
| 5 | **`7542c36d`** | `799640e6` | 0.57 | ✗ parse error: `expected ':'` (new .pto syntax) |
| 6 | **`7542c36d`** | `0659e29b` | 0.59 | ✗ assemble error: `'pto.tmrgsort' op format2 expects tmp.cols >= sum(src.cols) = 1024` |
| 7 | **`7542c36d`** | `0659e29b` | 0.60 | 🔴 **assembles; NaN logits for extents > 128** |

The seven commits between `42881d7b` and `7542c36d` are jointly exonerated (row 2 contains them all). The failure flips exactly at **`7542c36d` "Upgrade PTOAS to v0.60 and pin runtime to 0659e29 (#2523)"** — but that commit is an atomic bundle: its explicit-tmp `tcvt`/`tmrgsort` authoring changes (`python/pypto/ir/op/tile_ops.py`, `tensor_ops.py`, `language/op/*`) and PTOAS v0.60 **require each other** (rows 4–6), so old code only assembles on ≤0.57 and new code only on 0.60. The runtime (simpler) pin is not involved (row 3).

From a consumer (pypto-lib / pypto-serving) side the two halves of the bundle cannot be tested independently, so the regression is somewhere in the pair {new explicit-tmp lowering, PTOAS v0.60 assembly of it} — the pypto↔PTOAS split needs owner-side tooling (e.g. PTOAS 0.60 assembling a hand-written explicit-tmp repro, or pypto emitting the old-style form for a diff).

### Component

Codegen (explicit-tmp tcvt/tmrgsort lowering, introduced with the PTOAS v0.60 bump)

### Description

Since `7542c36d`, `l3_prefill_fwd`'s multi-tile walk corrupts outputs whenever a packed extent exceeds one 128-token tile — any second tile (including perfectly aligned ones: 256 = 2×128) produces NaN activations/logits. Requires real W8A8 weights (random-weight runs of the standalone harness don't show the serving-visible failure signature).

### Steps to Reproduce

Via pypto-serving (PR #204 head `a261960`; its KV-sizing ring-budget fix is required, otherwise a separate pooled-arena OOM masks this first):

```bash
pypto-serving \
  --model <w8a8 checkpoint dir> \
  --prompt "<' and' repeated 150 times> Huawei is" \
  --platform a2a3 --devices <8 free device ids> \
  --dp 8 --ep 8 --block-size 128 \
  --max-model-len 1024 --max-num-seqs 8 --max-num-batched-tokens 512 \
  --long-prefill-token-threshold 4096 \
  --generate-config '{"max_new_tokens": 16, "temperature": 0.0}' \
  --num-speculative-tokens 1 --no-enable-prefix-caching \
  --ring-dep-pool 131072 --ring-task-window 131072 --ring-heap 2147483648
```

Checkpoint: convert HF `deepseek-ai/DeepSeek-V4-Flash` with pypto-serving's `scripts/convert_deepseek_v4_to_w8a8.py`. Packed-extent sweep (prompt length): 64 ✅, 128 ✅, 129 ❌, 160 ❌, 256 ❌, 302 ❌ — deterministic across repeats.

The same flip is observable through pypto-lib 85e0790 alone: rows 2 vs 7 of the matrix above (its `models/deepseek_v4_flash_mtp/prefill_fwd.py` at `--num-tokens 160/302`), noting the stock golden harness skips validation and prints PASS unless outputs are dumped (see pypto-lib#1072).

### Expected Behavior

Correct greedy generation for any packed extent, as produced by `8cf3f058` + PTOAS v0.57 (rows 1–3).

### Actual Behavior

On `7542c36d`..main (`57e5c315`) + PTOAS v0.60:

```
WARNING | Sampler received non-finite logits
# ~3 s later, next program faults:
AICore 507015: "VEC instruction error: the ub address out of bounds (0x4000000000000000)"
# kernel: aicore_kernel_0_mix_aic (merged-binary KERNEL_ENTRY macro),
# task_id/func_id 15 -> kernel_config.py maps to qkv_proj_rope.py scopes
# (qr_proj_seed / qr_rms_norm_quant), launched with dynamic t_dim/T_TILE block count
```

The request then returns an empty completion; the lane is poisoned for subsequent requests.

### Git Commit ID

First bad: `7542c36dbfca87d36d2a1f5d3af1ac7a20a95f91` (reproduces through main tip `57e5c3153f3f8fbdfcc160e076a0becb3065e212`; parent `8cf3f05853a58b7745108acf45436029d302e948` is good)

### NPU Kind

Other: Ascend 910 — `a2a3` platform, 8 devices

### Host Platform

Linux (aarch64)

### Additional Context

Bisect environment (all pins consistent per commit):

| Component | Version (good rows) | Version (bad rows) |
|---|---|---|
| pypto | `8cf3f058` or `42881d7b` | `7542c36d` |
| simpler | `799640e6` (also tested `0659e29b`) | `0659e29b` |
| pto-isa | `cd4a3d3f` (pin of `799640e6`) / `be5ccb76` (pin of `0659e29b`) | `be5ccb76` |
| ptoas | 0.57 | 0.60 |
| pypto-lib | `85e0790` | `85e0790` |
| pypto-serving | PR #204 head `a261960` | PR #204 head `a261960` |
| CANN | 9.0.0 | 9.0.0 |

Related: pypto-lib#1072 (full symptom analysis, device-log mapping, dumped-output NaN locations), pypto-serving#204 (CI `k1-prefix-cache` is the red test). Happy to run further discriminating experiments on this machine (device access available) — e.g. assembling a specific pypto-emitted `.pto` under different ptoas builds if that helps split the pair.
